volatility3/framework/automagic/symbol_cache.py

```diff
@@ -22,164 +25,392 @@
-    def find_new_banner_files(cls, banners: Dict[bytes, List[str]], operating_system: str) -> List[str]:
-        """Gathers all files and remove existing banners"""
-        cacheables = list(intermed.IntermediateSymbolTable.file_symbol_url(operating_system))
-        for banner in banners:
-            for json_file in banners[banner]:
-                if json_file in cacheables:
-                    cacheables.remove(json_file)
-        return cacheables
+    def get_identifier(self, location: str) -> Optional[bytes]:
+        results = self._database.cursor().execute('SELECT identifier FROM cache WHERE location = ?',
+                                                  (location,)).fetchall()
+        for row in results:
+            return row['identifier']
+        return None
+
+    def get_location_statistics(self, location: str) -> Optional[Tuple[int, int, int, int]]:
+        results = self._database.cursor().execute(
+            'SELECT stats_base_types, stats_types, stats_enums, stats_symbols FROM cache WHERE location = ?',
+            (location,)).fetchall()
+        for row in results:
+            return row['stats_base_types'], row['stats_types'], row['stats_enums'], row['stats_symbols']
+        return None
+
+    def get_hash(self, location: str) -> Optional[str]:
+        results = self._database.cursor().execute('SELECT hash FROM cache WHERE location = ?',
+                                                  (location,)).fetchall()
+        for row in results:
+            return row['hash']
+
+    def update(self, progress_callback = None):
+        """Locates all files under the symbol directories.  Updates the cache with additions, modifications and removals.
+        This also updates remote locations based on a cache timeout.
+
+        """
+        on_disk_locations = set([filename for filename in intermed.IntermediateSymbolTable.file_symbol_url('')])
+        cached_locations = set(self.get_local_locations())
+
+        new_locations = on_disk_locations.difference(cached_locations)
+        missing_locations = cached_locations.difference(on_disk_locations)
+
+        cache_update = set()
+        files_to_timestamp = on_disk_locations.intersection(cached_locations)
+        if files_to_timestamp:
+            result = self._database.cursor().execute("SELECT location FROM cache WHERE local = True "
+                                                     f"AND cached < date('now', '{self.cache_period}');")
+            for row in result:
+                if row['location'] in files_to_timestamp:
+                    cache_update.add(row['location'])
+
+        idextractors = list(volatility3.framework.class_subclasses(IdentifierProcessor))
+
+        # New or not recently updated
+
+        counter = 0
+        files_to_process = new_locations.union(cache_update)
+        number_files_to_process = len(files_to_process)
+        cursor = self._database.cursor()
+        try:
+            for location in files_to_process:
```

Comment:
What do you think about using enumerate? It has the advantage of not having to declare a separate variable.
```suggestion
            for counter, location in enumerate(files_to_process):
```

Reply:
If you use this, you can remove the `counter` variable.